bin/long-to-linked

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3 
+"""
+Cut long reads and assign barcodes to the subreads. Optionally calculate span parameter for tigmint-cut.
+Usage: gunzip -c reads.fa.gz (or reads.fq.gz) | python3 long-to-linked -l length -m min_size -s auto_span -g genome_size | gzip > reads.cutlength.fa.gz
+@author: Janet Li
+"""
+import sys
+import os
+import argparse
+from read_fasta import read_fasta
+
+def split_long_read(read_string, length):
+    """Take an input read string and length, and cut the read into sub-reads of the specified length."""
+    return [read_string.strip()[0+i:length+i] for i in range(0, len(read_string), length)]
+
+def cut_reads(length, min_size, auto_span, genome_size, outfile):
+    """
+    Print cut reads to stdout. If auto_span is True, sequence coverage and span parameter are calculated,
+    and span is saved to tigmint-long.span.txt.
+    """
+    # Span automatically calculated as 1/4 of sequence coverage
+    cov_to_span = 0.25
+    total_bases = 0
+
+    with open("/dev/stdin", 'rt') as reads:
+        if not reads.isatty():  # Check that reads are being piped in from stdin
+            i = 0
+            for header, seq, _, _ in read_fasta(reads):
+                i += 1
+                read_length = len(seq)
+                total_bases += read_length
+                if read_length >= min_size:
+                    split_reads = split_long_read(seq, length)
+                    for i, read in enumerate(split_reads):
+                        print(">" + header + "_" + str(split_reads.index(read)) + " BX:Z:" + str(i), file=sys.stdout)
+                        print(read, file=sys.stdout)
+            if auto_span:
```

Comment:
Looks like this code will run for each line in the reads file? We only need to run it once, after all the reads have been gone through

Reply:
Sorry, do you mean to load all of the long reads first before splitting? Or are you referencing the `if auto_span` statement? That should be just running once after all of the long reads are read in and split (outside of the `for` loop).

Reply:
Aha yup you're totally right! My bad -- misread the spacing. You put it in the right place :)